source/Politics.cpp

```diff
@@ -226,6 +223,16 @@ string Politics::Fine(PlayerInfo &player, const Government *gov, int scan, const
 
 		if(!scan || (scan & ShipEvent::SCAN_CARGO))
 		{
+			// Illegal goods can be hidden inside legal goods to avoid detection.
+			const int contraband = ship->Cargo().IllegalCargoAmount();
+			const int concealment = ship->Attributes().Get("concealment");
+			const int legalGoods = ship->Cargo().Used() - contraband;
+
+			const int netIllegalCargo = contraband - concealment;
+			const double scaledLegalGoods = legalGoods * (1. + ship->Attributes().Get("scan interference"));
+			if(scaledLegalGoods && Random::Real() > 2. * netIllegalCargo / scaledLegalGoods)
```

Comment:
I just realized that this makes it so that having no legal goods means you always get caught regardless of your scan interference, which I don't think should be the case. Should it be that if you have no legal goods it uses the old formula? Or have it do one check against your scan interference then if that fails a second check against your cargo ratio? Or take the minimum between the old formula and the new formula (i.e. having >50% illegal goods doesn't hurt you any more than if you had exactly 50%)?